File: backend/app/schemas/llm_log.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

# ============================================================================
# Status enum
# ============================================================================

LlmLogStatus = Literal["ok", "not_configured", "error"]


# ============================================================================
# Summary (list view) — truncates user/response previews
# ============================================================================


class LlmRequestLogSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    prompt_name: str
    endpoint: str
    work_id: int | None = None
    status: LlmLogStatus
    duration_ms: int
    model: str | None = None
    provider: str | None = None
    profile_id: int | None = None
    prompt_assembly_id: int | None = None
    user_preview: str = ""
    response_preview: str = ""
    error: str | None = None
    created_at: datetime


# ============================================================================
# Detail (single log) — full system / user / response
# ============================================================================


class LlmRequestLogDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    prompt_name: str
    endpoint: str
    work_id: int | None = None
    system: str
    user: str
    response: str | None = None
    status: LlmLogStatus
    error: str | None = None
    duration_ms: int
    model: str | None = None
    provider: str | None = None
    profile_id: int | None = None
    prompt_assembly_id: int | None = None
    created_at: datetime
    updated_at: datetime


# ============================================================================
# List response — paginated
# ============================================================================


class LlmRequestLogList(BaseModel):
    items: list[LlmRequestLogSummary]
    total: int
    page: int = Field(1, ge=1)
    page_size: int = Field(20, ge=1, le=200)

File: frontend/src/types/llm-log.ts
export type LlmLogStatus = "ok" | "not_configured" | "error";

export interface LlmRequestLogSummary {
  id: number;
  prompt_name: string;
  endpoint: string;
  work_id: number | null;
  status: LlmLogStatus;
  duration_ms: number;
  model: string | null;
  provider: string | null;
  profile_id: number | null;
  prompt_assembly_id: number | null;
  user_preview: string;
  response_preview: string;
  error: string | null;
  created_at: string;
}

export interface LlmRequestLogDetail extends LlmRequestLogSummary {
  system: string;
  user: string;
  response: string | null;
  updated_at: string;
}

export interface LlmRequestLogList {
  items: LlmRequestLogSummary[];
  total: number;
  page: number;
  page_size: number;
}

export const LLM_LOG_STATUS_LABEL: Record<LlmLogStatus, string> = {
  ok: "成功",
  not_configured: "未配置",
  error: "失败",
};

export const LLM_LOG_STATUS_COLOR: Record<LlmLogStatus, string> = {
  ok: "green",
  not_configured: "gold",
  error: "red",
};

export const LLM_LOG_STATUS_OPTIONS: { value: LlmLogStatus; label: string }[] = [
  { value: "ok", label: LLM_LOG_STATUS_LABEL.ok },
  { value: "not_configured", label: LLM_LOG_STATUS_LABEL.not_configured },
  { value: "error", label: LLM_LOG_STATUS_LABEL.error },
];
